### PR TITLE
Refactor `MarketDataService` logic for better latency and scalability

### PR DESCRIPTION
**Pull request by leonardoarcari**
## Pull request details

Massive refactoring of backend code structure. Most relevant changes are outlined as follows:

- Significant project folder refactoring to better highlight the backend Hexagonal architecture 
- `MarketDataService` no longer lazily fetches price updates, but returns whatever is stored at the time of `get_market()` call. This is expected to significantly reduce REST endpoints latency and jitter
- The new `PriceUpdaterWorker` periodically refreshes market prices and updates `MarketDataService`
- `Maintenance` Tokio task has been renamed to `MarketDiscoveryWorker` and daily checks for new crypto assets and markets, price included.
- Bump Rust dependencies

## Breaking changes

_None_

## Issues fixed

1. #206

### CODE REVIEW

File: dcapal-backend/docs/rest/public/import/post.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Import portfolio
+
+Temporarely upload DcaPal portfolio definition to later import it in DcaPal client
```
Comment (mattiapitossi) [inline dcapal-backend/docs/rest/public/import/post.md:3]:
```suggestion
Temporarily upload DcaPal portfolio definition to later import it in DcaPal client
```

Comment (leonardoarcari) [reply dcapal-backend/docs/rest/public/import/post.md]: Thanks fixed!

File: dcapal-backend/docs/rest/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+In this section you can find DcaPal REST APIs intended for Developers, wishing to integrate their tools or apps with DcaPal.
+
+### Open endpoints
+
+#### Import Portfolio
+
+Developers can integrate with DcaPal importing their portfolios via REST API and later open it in the client
+
+- [Import portfolio](public/import/post.md): `POST /import/portfolio`
```
[the remark this answers is not in the document]
Comment (mattiapitossi) [reply dcapal-backend/docs/rest/README.md]: I remember that we had a conversation about moving this project to a public organization, so we can also take advantage of moving the project there to publish the documentation as well.

**Comment by mattiapitossi** [comment]
LGTM!

**Review by mattiapitossi — APPROVED** [review]

**Pull request merged**